src/Core/Geographic/Crs.ts

```diff
@@ -55,22 +60,22 @@ function _unitFromProj4Unit(projunit) {
     }
 }
 
-function toUnit(crs) {
+function toUnit(crs: ProjectionLike) {
     mustBeString(crs);
     switch (crs) {
         case 'EPSG:4326' : return UNIT.DEGREE;
         case 'EPSG:4978' : return UNIT.METER;
         default: {
             const p = proj4.defs(formatToEPSG(crs));
-            if (!p) {
+            if (!p?.units) {
                 return undefined;
             }
             return _unitFromProj4Unit(p.units);
         }
     }
 }
 
-function toUnitWithError(crs) {
+function toUnitWithError(crs: ProjectionLike) {
```

Comment:
Maybe we could rename it to a more explicit name, like `checkUnit` ? I admit I don't understand the basic role of the method without reading its code.

Reply:
The method instead of returning `unit | undefined` returns `unit` and throws if `undefined`. In my opinion, we should add a method to check whether a `proj4.defs` is defined.